cli/medperf/commands/dataset/submit.py

```diff
@@ -43,11 +43,17 @@ def run(
             submit_as_prepared,
             for_test,
         )
-        preparation.validate()
-        preparation.validate_prep_cube()
-        preparation.create_dataset_object()
-        if submit_as_prepared:
-            preparation.make_dataset_prepared()
+        submission_dict = preparation.prepare_dict(submit_as_prepared)
+        config.ui.print(dict_pretty_format(submission_dict))
+
+        msg = "Do you approve the registration of the presented data to MedPerf? [Y/n] "
+        warning = (
+            "Upon submission, your email address will be visible to the Data Preparation"
+            + " Owner for traceability and debugging purposes."
+        )
+        config.ui.print_warning(warning)
+        preparation.approved = preparation.approved or approval_prompt(msg)
```

Comment:
It would be better to instead convert this logic into a function, and keep the original flow with the added new function. Just so that we keep a consistent style where you can get an idea of the workflow of a command by looking at the `run` function.